Wrappers/Python/cil/optimisation/operators/Operator.py

```diff
@@ -187,31 +187,34 @@ def dot_test(operator, domain_init=None, range_init=None, verbose=False, **kwarg
         :param domain_init: optional initialisation container in the operator domain 
         :returns: boolean, True if the test is passed.
         :param decimal: desired precision
-        :type decimal: int, optional, default 4
+        :type decimal: int, optional, default 4        
         '''
-        seed = kwargs.get('seed',None)
-        if range_init is None:
-            y = operator.range_geometry().allocate('random', seed=seed)
-        else:
-            y = range_init
-        if domain_init is None:
-            x = operator.domain_geometry().allocate('random',seed=seed)
-        else:
-            x = domain_init
+
+        lst = [1,2,3]

```

Comment:
If A^T is really the transpose of A it'll always pass, but it may pass in other cases too. We might happen to test with all zeros, or other cases where the errors cancel out. But I don't think the solution is to test more values, I think if you want a comprehensive test you have to inspect every element of A and A^T in turn and compare them. Otherwise we have the dottest with a seed, and if we want to test many values we do that by calling dottest many times.





I guess the true test wouldto set each  each voxel in turn to

Reply:
this comment is cut @gfardell ?

Reply:
:D you'll never know!

no, it was what I typed above about inspecting each voxel as you set it and call direct/adjoint. Just bad editing from me!